addons/portal/controllers/mail.py

```diff
@@ -37,49 +40,65 @@ def portal_avatar(self, res_id=None, height=50, width=50, access_token=None, _ha
         )
         return stream.get_response()
 
-    @http.route('/mail/chatter_init', type='json', auth='public', website=True)
-    def portal_chatter_init(self, res_model, res_id, domain=False, limit=False, **kwargs):
-        is_user_public = request.env.user._is_public()
-        message_data = self.portal_message_fetch(res_model, res_id, domain=domain, limit=limit, **kwargs)
-        display_composer = False
-        if kwargs.get('allow_composer'):
-            display_composer = kwargs.get('token') or not is_user_public
-        return {
-            'messages': message_data['messages'],
-            'options': {
-                'message_count': message_data['message_count'],
-                'is_user_public': is_user_public,
-                'is_user_employee': request.env.user._is_internal(),
-                'is_user_publisher': request.env.user.has_group('website.group_website_restricted_editor'),
-                'display_composer': display_composer,
-                'partner_id': request.env.user.partner_id.id
+    @http.route("/portal/chatter_init", type="json", auth="public", website=True)
+    def portal_chatter_init(self, thread_model, thread_id, **kwargs):
+        store = Store()
+        thread = request.env[thread_model]._get_thread_with_access(thread_id, **kwargs)
+        if not thread:
+            raise Forbidden()
+        partner = request.env.user.partner_id
+        if request.env.user._is_public():
+            with suppress(AccessError):
+                if portal_partner := get_portal_partner(
+                    thread, kwargs.get("hash"), kwargs.get("pid"), kwargs.get("token")
+                ):
+                    partner = portal_partner
+        store.add(
+            {
+                "self": Store.one(
+                    partner,
+                    fields=["active", "isAdmin", "name", "user", "write_date"],
+                ),
```

Comment:
What about guest?

We ideally don't want to ever put public user as `self` in store.

If there is no portal partner (user is public) and no guest, it would be better to raise instead.
Can this happen in some flows?